src/stream/src/executor/source/source_executor.rs

```diff
@@ -211,16 +215,16 @@ impl<S: StateStore> SourceExecutor<S> {
                     split.clone()
                 };
 
-                core.state_cache
+                core.updated_splits_in_epoch
                     .entry(split.id())
                     .or_insert_with(|| initial_state.clone());
 
                 target_state.push(initial_state);
             }
         }
 
-        // state cache may be stale
-        for existing_split_id in core.stream_source_splits.keys() {
+        // Checks dropped splits
+        for existing_split_id in core.latest_split_info.keys() {
```

Comment:
also delete removed items in `latest_split_info` here because it is seen as the ground truth of split assignment when doing recovery internally.

Reply:
Actually it is updated later in `persist_state_and_clear_cache` using `target_state`.. This is how it works previously. This PR doesn't want to change it. https://github.com/risingwavelabs/risingwave/pull/15104 just changes this.